### ISSUE TITLE
Detect comparisons with literal values that are always false

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1674]
It would help find a certain class of bugs if pyright detected and reported the case where an if statement is comparing against a literal value (string, number, enum) and there is no way that condition will ever be true.

```python
OS = Literal['Linux', 'Darwin', 'Windows']
os: OS = 'Linux'

# Sometime later
if os == 'linux':
    ...
```

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.124, which I just published. It will also be included in the next release of pylance.

**Comment by bl-ue** [comment]
For reference: https://github.com/microsoft/pyright/discussions/1671

**Comment by Avishayy** [comment]
This does not work for me, I'm using pyright 1.1.230 and no diagnostic is generated from the sample in the thread. Does this require setting a flag in the configuration? (Like mypy does with [strict_equality](https://mypy.readthedocs.io/en/stable/config_file.html#confval-strict_equality))

I couldn't find any docs on this.

**Comment by erictraut** [comment]
You need to enable the `reportUnnecessaryComparison` diagnostic check. It is not enabled by default. It is part of `strict` mode. For full documentation of configuration options, refer to [these docs](https://github.com/microsoft/pyright/blob/main/docs/configuration.md).

**Comment by Avishayy** [comment]
Thanks! I tried to find the setting by searching "literal" but couldn't find it.

By the way, if I reverse the condition (`!=` instead of `==`) the diagnostic still exists (as expected), but the message is wrong
![image](https://user-images.githubusercontent.com/5567310/170856969-6e72763d-d4d9-4ede-bb5c-5299f1d6d40e.png)
as it will always evaluate to True in that case.

**Comment by erictraut** [comment]
Ah, good catch. This will be fixed in the next release.
